timely/src/progress/broadcast.rs

```diff
@@ -43,8 +43,8 @@ impl<T:Timestamp+Send> Progcaster<T> {
         let addr = path.clone();
         Progcaster {
             to_push: None,
-            pushers,
-            puller,
+            pushers: pushers.into_iter().map(|p| Box::new(p) as Box<dyn Push<_>>).collect(),
```

Comment:
@frankmcsherry This is the only place I kept the trait objects because otherwise `Progcaster` would have to become generic over the pusher which would then add a generic parameter to `Subgraph`. I didn't know if that was ok but if it is I can eliminate this last bit of trait object if that sounds good